src/mixer/mod.rs

```diff
@@ -0,0 +1,26 @@
+use std::borrow::Cow;
+
+pub mod softmixer;
+
+pub trait Mixer {
+    fn init(&self);
+    fn start(&self);
+    fn stop(&self);
+    fn set_volume(&self, volume: u16);
+    fn volume(&self) -> u16;
+    fn get_stream_editor(&self) -> Option<Box<StreamEditor + Send>>
+    {
+        None
+    }
+}
+
+pub trait StreamEditor {
+  fn modify_stream<'a>(&self, data: &'a [i16]) -> Cow<'a, [i16]>;
```

Comment:
How can I do that without having issues with the object safety of the AudioFilter trait?
let buffer = if let Some(ref editor) = stream_editor {
^^^^^^ the trait `std::marker::Sized` is not implemented for `[i16]`